sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessage.java

```diff
@@ -234,8 +250,8 @@ public BinaryData getBody() {
                 return BinaryData.fromBytes(amqpAnnotatedMessage.getBody().getFirstData());
             case SEQUENCE:
             case VALUE:
-                throw logger.logExceptionAsError(new UnsupportedOperationException("Not supported AmqpBodyType: "
-                    + type.toString()));
+                throw logger.logExceptionAsError(new IllegalStateException("Message  body type is not DATA, instead " +
+                    "it is: " + type.toString()));
```

Comment:
Now that we have added support for SEQUENCE and VALUE, why do we still throw an exception here?

Reply:
good catch.